injected/src/features/duckplayer-native/messages.js

```diff
@@ -0,0 +1,119 @@
+import * as constants from './constants.js';
+import { mockTransport } from './mock-transport.js';
+
+/** @import {YouTubeError} from './error-detection.js' */
+/** @import {Environment} from './environment.js' */
+
+/**
+ * @typedef {object} MuteSettings - Settings passed to the onMute callback
+ * @property {boolean} mute - Set to true to mute the video, false to unmute
+ */
+
+/**
+ * @typedef {object} MediaControlSettings - Settings passed to the onMediaControll callback
+ * @property {boolean} pause - Set to true to pause the video, false to play
+ */
+
+/**
+ * @typedef {object} UrlChangeSettings - Settings passed to the onURLChange callback
+ * @property {PageType} pageType
+ */
+
+/**
+ * @typedef {'UNKNOWN'|'YOUTUBE'|'NOCOOKIE'|'SERP'} PageType
+ */
+
+/**
+ * @import {Messaging} from '@duckduckgo/messaging'
+ *
+ * A wrapper for all communications.
+ *
+ * Please see https://duckduckgo.github.io/content-scope-utils/modules/Webkit_Messaging for the underlying
+ * messaging primitives.
+ */
+export class DuckPlayerNativeMessages {
+    /**
+     * @param {Messaging} messaging
+     * @param {Environment} environment
+     * @internal
+     */
+    constructor(messaging, environment) {
+        /**
+         * @internal
+         */
+        this.messaging = messaging;
+        this.environment = environment;
+
+        if (this.environment.isIntegrationMode()) {
+            this.messaging.transport = mockTransport();
```

Comment:
@shakyShane Is there a better way to patch in mockTransport here? dropLabels maybe?